datasets/swiss_judgment_prediction/swiss_judgment_prediction.py

```diff
@@ -119,56 +136,62 @@ def _split_generators(self, dl_manager):
-
-        if self.config.language == "all_languages":
+        if self.config.language == "all":  # without mt pseudo languages
             with open(filepath, encoding="utf-8") as f:
                 for id_, row in enumerate(f):
                     data = json.loads(row)
-                    yield id_, {
-                        "id": data["id"],
-                        "year": data["year"],
-                        "text": data["text"],
-                        "label": data["label"],
-                        "language": data["language"],
-                        "region": data["region"],
-                        "canton": data["canton"],
-                        "legal area": data["legal area"],
-                    }
-        else:
+                    _ = data.setdefault("source_language", "n/a")
+                    yield id_, data
+        elif self.config.language == "all+mt":  # original languages and mt languages
+            with open(filepath, encoding="utf-8") as f:
+                for id_, row in enumerate(f):
+                    data = json.loads(row)
+                    _ = data.setdefault("source_language", "n/a")
+                    yield id_, data
+            last_id = id_ + 1
+            if mt_filepath:  # yield data from mt_filepath
+                with open(mt_filepath, encoding="utf-8") as f:
+                    for id_, row in enumerate(f):
+                        data = json.loads(row)
+                        _ = data.setdefault("source_language", "n/a")
+                        yield last_id + id_, data
+        elif self.config.language in _MT_LANGUAGES:  # mt languages
+            if mt_filepath:  # yield data from mt_filepath
+                with open(mt_filepath, encoding="utf-8") as f:
+                    for id_, row in enumerate(f):
+                        data = json.loads(row)
+                        _ = data.setdefault("source_language", "n/a")
+                        if data["language"] in self.config.language:  # "de" in "mt_de"
+                            yield id_, data
+        else:  # original languages
+            assert self.config.language in _ORIGINAL_LANGUAGES
             with open(filepath, encoding="utf-8") as f:
                 for id_, row in enumerate(f):
                     data = json.loads(row)
+                    _ = data.setdefault("source_language", "n/a")
                     if data["language"] == self.config.language:
-                        yield id_, {
-                            "id": data["id"],
-                            "year": data["year"],
-                            "text": data["text"],
-                            "label": data["label"],
-                            "language": data["language"],
-                            "region": data["region"],
-                            "canton": data["canton"],
-                            "legal area": data["legal area"],
-                        }
+                        yield id_, data
```

Comment:
My proposal to simplify the `_generate_examples` method:
```python
    def _generate_examples(self, filepath, mt_filepath):
        """This function returns the examples in the raw (text) form."""
        if self.config.language in ["all", "all+mt"] + _ORIGINAL_LANGUAGES:
            with open(filepath, encoding="utf-8") as f:
                for id_, row in enumerate(f):
                    data = json.loads(row)
                    _ = data.setdefault("source_language", "n/a")
                    if self.config.language in ["all", "all+mt"] or data["language"] == self.config.language:
                        yield id_, data
        if self.config.language in ["all+mt"] + _MT_LANGUAGES:
            if mt_filepath:  # yield data from mt_filepath
                with open(mt_filepath, encoding="utf-8") as f:
                    for id_, row in enumerate(f):
                        data = json.loads(row)
                        _ = data.setdefault("source_language", "n/a")
                        if (
                            self.config.language == "all+mt" or data["language"] in self.config.language
                        ):  # "de" in "mt_de"
                            yield f"mt_{id_}", data
```

Reply:
Thanks for the proposal. Yes, this would simplify it. However, it would introduce an inconsistency in the ids, right?
For "filepath" it will be an int, and for "mt_filepath" it will be a string

Reply:
You are right, I found an inconsistency, but related to the "if" conditions. Let me try to fix my proposal.

Reply:
Aah I see. Thanks for the proposal, I integrated it.